shared/src/packages/surveys/libs/validation-schemas/survey.validation-schema.ts

```diff
@@ -0,0 +1,30 @@
+import joi from 'joi';
+
+import { PREFERENCES_OTHER_CATEGORY } from '../constants/constants.js';
+import { SurveyValidationMessage } from '../enums/enums.js';
+import { type SurveyInputDto } from '../types/types.js';
+
+const MINIMUM_ARRAY_LENGTH = 1;
+
+const surveyInput = joi.object<SurveyInputDto, true>({
+  preferences: joi
+    .array()
+    .items(joi.string())
+    .min(MINIMUM_ARRAY_LENGTH)
+    .messages({
+      'array.min': SurveyValidationMessage.OPTION_REQUIRED,
+    }),
+  other: joi.string().when('preferences', {
+    is: joi
+      .array()
+      .has(joi.string().valid(PREFERENCES_OTHER_CATEGORY))
+      .min(MINIMUM_ARRAY_LENGTH),
+    then: joi.string().trim().required().messages({
```

Comment:
we should add max length for other here, and also instead of MINIMUM_ARRAY_LENGTH and max length for textarea use values from CreateSurveyValidationRule enum, it's already created. And as @v0ldemar01 said we need to use  CreateSurveyValidationRule instead of max textarea length on mobile, we can fully remove this prop from input as we will handle this using validation